### ISSUE TITLE
http protocol is hardcoded

### ISSUE THREAD

**Issue by imanzano** [kamon-io/kamon-influxdb#22]
We are not able to setup https protocol.

https://github.com/kamon-io/kamon-influxdb/blob/a2bfbbd163db872bffa59d952869172bf538f5f4/src/main/scala/kamon/influxdb/InfluxDBReporter.scala#L153

**Comment by Qthunder** [comment]
Am working on a PR for this (the change is a few lines). Am having trouble writing the test using the `MockWebServer`

**Comment by Qthunder** [comment]
Opened PR https://github.com/kamon-io/kamon-influxdb/pull/24

**Comment by martin-g** [comment]
+1

**Comment by martin-g** [comment]
Here is my brave hack until this fix is released:

```scala
class MyInfluxDBReporter extends InfluxDBReporter {
  override def start(): Unit = {
    https()
    super.start()
  }
  
  override def reconfigure(config: config.Config): Unit = {
    super.reconfigure(config)
    https()
  }

  private def https(): Unit = {
    val settingsField = classOf[InfluxDBReporter].getDeclaredField("settings")
    settingsField.setAccessible(true)
    val settings = settingsField.get(this)
    val urlField = settings.getClass.getDeclaredField("url")
    urlField.setAccessible(true)
    val httpUrl = urlField.get(settings).toString
    val httpsUrl = httpUrl.replaceFirst("http://", "https://")
    urlField.set(settings, httpsUrl)
  }
}
```

**Comment by martin-g** [comment]
This is not enough unfortunately because in my case I need to setup SSL certificates for OkHttpClient.
Can we please make kamon.influxdb.InfluxDBReporter#buildClient() `protected` too ?

**Comment by dpsoft** [comment]
closed via #24
